### DISCUSSION TITLE
root-relative links and github-pages

### DISCUSSION THREAD

**Discussion by madeinoz67** [squidfunk/mkdocs-material#3126] [Q&A] [answered]
In my docs I'm creating all internal links as root-relative, e.g. a link form one page to something else as e.g. `[link](/topic)` however when published to GitHub pages these links can't be found as per how git-hub pages adds the project name into path of the URL. e.g. https://username.github.io/project/topic

Whats the best way of handling root-relative links in mkdocs-materials?

Cheers

**Comment by squidfunk** [comment, marked as answer]
Have you defined `site_url`? You should set `site_url` to the root path of your documentation project:

https://github.com/squidfunk/mkdocs-material/blob/ff53ba0d76bd2adf40fab3053613ef0ea8ae67a1/mkdocs.yml#L23

If that doesn't work, there might be [some plugins](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins#links--references) that help to transform root-relative links prior to the build. Another idea is to use the `macros` plugin: https://github.com/mkdocs/mkdocs/issues/192#issuecomment-771630318

**Reply by madeinoz67** [reply]
Thanks for that, I added site_url and didnt make any differnce, in the end what worked for me is to use the ezlinks plugin <https://github.com/orbikm/mkdocs-ezlinks-plugin>, while not root-relative links it was good enough for what I wanted.
